modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/outgoing/OutgoingSnapshotsManager.java

```diff
@@ -0,0 +1,145 @@
+import org.apache.ignite.internal.table.message.SnapshotMetaRequest;
+import org.apache.ignite.internal.table.message.SnapshotMvDataRequest;
+import org.apache.ignite.internal.table.message.SnapshotRequestMessage;
+import org.apache.ignite.internal.table.message.SnapshotTxDataRequest;
+import org.apache.ignite.internal.table.message.TableMessageGroup;
+import org.apache.ignite.network.MessagingService;
+import org.apache.ignite.network.NetworkAddress;
+import org.apache.ignite.network.NetworkMessage;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Outgoing snapshots manager.
+ */
+public class OutgoingSnapshotsManager implements IgniteComponent {
+    /** Logger. */
+    private static final IgniteLogger LOG = Loggers.forClass(OutgoingSnapshotsManager.class);
+
+    /** Messaging service. */
+    private final MessagingService messagingService;
+
+    /** Map with outgoing snapshots. */
+    private final ConcurrentMap<UUID, OutgoingSnapshot> outgoingSnapshots = new ConcurrentHashMap<>();
+
+    /**
+     * Constructor.
+     *
+     * @param messagingService Messaging service.
+     */
+    public OutgoingSnapshotsManager(MessagingService messagingService) {
+        this.messagingService = messagingService;
+    }
+
+    /**
+     * Returns a messagins service.
+     */
+    public MessagingService messagingService() {
+        return messagingService;
+    }
+
+    @Override
+    public void start() {
+        messagingService.addMessageHandler(TableMessageGroup.class, this::messageHandler);
+    }
+
+    @Override
+    public void stop() throws Exception {
+    }
+
+    /**
+     * Registers an outgoing snapshot in the manager.
+     *
+     * @param snapshotId Snapshot id.
+     * @param outgoingSnapshot Outgoing snapshot.
+     */
+    void startOutgoingSnapshot(UUID snapshotId, OutgoingSnapshot outgoingSnapshot) {
+        outgoingSnapshots.put(snapshotId, outgoingSnapshot);
+    }
+
+    /**
+     * Removes an outgoing snapshot from the manager.
+     *
+     * @param snapshotId Snapshot id.
+     */
+    void stopOutgoingSnapshot(UUID snapshotId) {
+        outgoingSnapshots.remove(snapshotId);
+    }
+
+    private void messageHandler(NetworkMessage networkMessage, NetworkAddress sender, @Nullable Long correlationId) {
+        // Ignore all messages that we can't handle.
+        if (!(networkMessage instanceof SnapshotRequestMessage)) {
+            return;
+        }
+
+        assert correlationId != null;
+
+        OutgoingSnapshot outgoingSnapshot = outgoingSnapshots.get(((SnapshotRequestMessage) networkMessage).id());
+
+        if (outgoingSnapshot == null) {
+            if (LOG.isWarnEnabled()) {
+                LOG.warn("Unexpected snapshot request message has been received [message={}]", networkMessage.toString());
+            }
+
+            return;
+        }
+
+        CompletableFuture<? extends NetworkMessage> responseFuture;
+
+        switch (networkMessage.messageType()) {
```

Comment:
I would suggest to extract this into a separate method

Reply:
Come on, it's not that big

Reply:
I stand by my suggestion)

Reply:
I did the extraction